Week-6/Assignment-3/index.js

```diff
@@ -0,0 +1,122 @@
+const arr = [0, 2, 1, 1, 0, 1, 2, 1];
+const arr2 = [0, 2, 1, 0, 1, 0, 1, 2, 1, 2, 2, 1, 0, 1, 2, 1];
+const arr3 = [0, 1, 2, 3, 4];
+
+/**
+ * @function validateInput validates if the provided input is an array & has all 0s, 1s & 2s
+ * @param {Array} arr an array
+ * @returns boolean
+ */
+const validateInput = (arr) => Array.isArray(arr) &&
+    arr.every(el => Number.isInteger(el) && (el === 0 || el === 1 || el === 2));
+
+/**
+ * @function sortArr a function which sorts array of 0s, 1s & 2s
+ * @summary Time Compexity - O(n), Space Complexity - O(1)
```

Comment:
The space complexity here is O(n).

Reply:
Thanks @sairamin, got it, I've merged it under - https://github.com/pesto-students/p3-sai-tusharwalzade216/pull/38